src/app/index.ts

```diff
@@ -0,0 +1,1003 @@
+        this.name = name;
+        this.year = year;
+        this.price = price;
+        this.amount = amount;
+    }
+
+    substractAmount(quantity: number) {
+        if (this.amount !== null) {
+            this.amount -= quantity;
+        }
+
+    }
+}
+
+class User {
+    public id: string;
+    public name: string;
+    public money: number;
+    public orders: any;
+    public cart: any;
+    constructor(name?: string, money?: number) {
+        this.id = `u${Date.now()}`;
+
+        this.name = name ? name : '';
+
+        this.money = money ? money : 0;
+
+
+        this.orders = [];
+        this.cart = this.getCart();
+    }
+
+    showOrderHistory(direction = 1, field = 'date') {
+        return this.orders.sort((a: any, b: any) => {
+            if (a[field] > b[field]) {
+                return direction;
+            }
+            if (a[field] < b[field]) {
+                return -direction;
+            }
+
+            return 0;
+        });
+    }
+
+    getCart() {
+        return new Cart();
+    }
+
+    addOrder() {
+        this.orders.push(new Order(this.cart.goods, this.cart.totalSum));
+    }
+}
+
+class Admin extends User {
+    public obj: any;
+    constructor(obj: any) {
+        super(obj, 1);
+    }
+
+    createProduct(name: string, year: number, price: number, amount: number) {
+        products.push(new Product(name, year, price, amount));
+    }
+}
+
+const cartCount = document.querySelector(".cart__count");
+const cartLink = document.querySelector(".cart__btn");
+const makeOrder = document.querySelector(".make-order-btn");
+const cartBody = document.querySelector(".cart-items tbody");
+const errorMsg = document.querySelector('.make-order > .error');
+const searchButton = document.querySelector('.button_for_search');
+const searchInput = document.querySelector('.place_for_search');
+const clearButton = document.querySelector('.button_for_turn_back');
+
+const btnPhone = document.querySelector('.nav__link.phone');
+const btnAll = document.querySelector('.nav__link.all');
+const btnWatch = document.querySelector('.nav__link.watch');
+const btnAir = document.querySelector('.nav__link.air');
+const btnMac = document.querySelector('.nav__link.mac');
+
+let afterOrder = false;
+
+function setCardCount() {
+    if (currentUser) {
+        (cartCount as Element).innerHTML = currentUser.cart.size;
+        if (currentUser.cart.size) {
+            (cartCount as HTMLElement).style.display = "inline-flex";
+        } else {
+            (cartCount as HTMLElement).style.display = "none";
+        }
+        checkCartSum();
+    }
+}
+
+function checkCartSum() {
+    (document.querySelector(".header-price .price") as Element).innerHTML =
+        currentUser.cart.totalSum;
+}
+
+function clearCart() {
+    (cartBody as Element).innerHTML = "";
+}
+
+(cartLink as Element).addEventListener("click", renderCart);
+
+
+
+(makeOrder as HTMLElement).onclick = function (event: any) {
```

Comment:
it's good practice to use one approach for listening to events. Of course, if it's really necessary you can use all the features of the language